src/plugins/macaddr/macaddr.c

```diff
@@ -160,6 +160,7 @@ int elektraMacaddrGet (Plugin * handle ELEKTRA_UNUSED, KeySet * returned, Key *
 
 		if (rc != VALIDATION_ISINT)
 		{
+			keySetMeta (cur, "origvalue", keyString (cur));
```

Comment:
You need to set `origvalue` after calling `transformMac`, because `transformMac` calls `keySetString` which resets `origvalue`.

To detect this problem in future, see the comments in `testmod_macaddr.c`